lib/internal/Magento/Framework/Interception/Code/InterfaceValidator.php

```diff
@@ -205,12 +202,14 @@ protected function getOriginMethodName($pluginMethodName)
      */
     protected function getMethodType($pluginMethodName)
     {
-        if (substr($pluginMethodName, 0, 6) == self::METHOD_BEFORE) {
+        if (0 === strpos($pluginMethodName, self::METHOD_AFTER)) {
```

Comment:
@lbajsarowicz please avoid Yoda conditions.

Reply:
What's the reason to avoid Yoda condition?

Reply:
Readability and consistency. We are writing code for human beings and considering amount of automation we have there is no need to write code in such unnatural way.

I'm going to borrow a rule for this from `php-cs-fixer` but until it is not fixed all over the codebase there is no need to increase amount of such occurrences.

Reply:
During my CRs I've seen several occurences when someone "lost" the `=` in comparison, which resulted in assignment instead of comparison. That's where Yoda condition is just priceless.

If you don't like Yoda condition - then create not 1 but 2 rules:
1. Forbid Yoda condition
2. Forbid assignment in all condition instructions

Reply:
Assignment in condition is totally fine. We should not deliver any untested code (at least manually) in the first place.

Reply:
I cannot say how your brain works but for me and most humans reading from left to right
`if (strpos($pluginMethodName, self::METHOD_BEFORE) === 0) {`
is much easier to comprehend.

In Yoda variant you meet zero, then think "what zero?", read right part of the condition and then have to go back to zero anyway, this time you will understand what was its meaning.

Reply:
> people drop parenthesis in random places

Yeah, those should be stripped out. I think I met such fixers before as well.

Reply:
I can't find the example right now, but I fixed at least 5 occurrences of such:
```php
if(strpos('oh, oh. i are the developer', 'am i super magento developer?!' === 0)) {
...
}
```
in this case with Yoda, you just can't make this stupid mistake.

Regarding reading left to right reading - I scan the line starting from left and seeing `0 === strpos(` I know that I expect string to start with other string. The same with `false === strpos` I expect that one string does not contain another. First I focus on the expectation, then on details that are in the body of method call.

Reply:
Yoda conditions is  populat aproach all around the globe. I used it in my daily coding standards.

@orlangur  Wonder if that reluctance for yoda is your personal thing...or is it somewhere in [M2 coding standard](https://devdocs.magento.com/guides/v2.3/coding-standards/bk-coding-standards.html) ?
I could not find it.

Reply:
Developer should always check **entire** `if` statement when checking the code. I personally think that Yoda condition is absolutely fine and there are no issues with it. It also improves the overall quality of the code and allows you to avoid a lot of mistakes as @lbajsarowicz mentioned. 

If you want to avoid Yoda conditions, make it an official standard for Magento contributions.

Reply:
As for me both variants are ok.

Some time ago I also didn’t like yoga style comparisons, but after finding few mistakes like @lbajsarowicz showed above I think it’s better to have it.

Reply:
I'm for Yoda conditions 🙌 This approach definitely should not be discouraged

Reply:
Both options are ok. 
We dont have any strict rules that prohibit this now.

Reply:
@adeptofvoltron @qsolutions-pl @roma-glushko
Hi guys, this is something not enforced yet but it definitely should IMO to not break ones eyes as I mentioned above.

```
if ('Luke Skywalker' === $name) {
}
```

> In english, this translates to "If Luke Skywalker is the name", which should feel very familiar if you are from Dagobah or whichever planet Yoda actually came from, but is just cognitive load if you are used to english syntax, which you should if you are writing programs.

Please check https://github.com/magento/magento2/pull/27903#discussion_r420018472 and provide arguments for using Yoda.

@lbajsarowicz @ihor-sviziev `if (strpos('needle', 'haystack'))` is not always a mistake, actually - it can be used when we know `needle` is not in the very beginning of `haystack`. And even if it was always a mistake, it would be better to write some rule for static analyzer producing warning in such case than spoil every single condition for human beings.

Writing less readable code to solve some potential problem is simply an improper solution. Mistakes should be avoided by thorough testing, preferably automated. Luckily Magento codebase is mostly readable currently and it will be definitely a move forward to make it fully consistent.

Reply:
@orlangur The example you provided with `Luke Skywalker` is a completely different type of comparison and is not the subject of this discussion. Thank for all your efforts, but I'm still going to use `0 === strpos` or `false === strpos` when it's necessary to use these explicitly.

Reply:
@lbajsarowicz "zero is equal to string position" how it is different?